src/utils/fs.ts

```diff
@@ -0,0 +1,65 @@
+import { readFile, stat } from 'fs/promises';
+
+/**
+ * Convert paths with backslashes to normalized paths with forward slashes.
+ *
+ * Extended-length paths on Windows (starting with \\\\?\\) are not normalized due to
+ * exceeding the MAX_PATH (260 characters). They need this prefix so that Windows can
+ * identify them as an extended-length path.
+ *
+ * This is useful when building a project on Windows as the path names in the Next.js
+ * build output middleware manifest are in the forward slash format, while Windows
+ * uses backslashes.
+ *
+ * @param path A path with backslashes.
+ * @returns A path with forward slashes.
+ *
+ * @example
+ * ```ts
+ * const normalized = normalizePath("D:\\path\\with\\backslashes");
+ * // normalized === "D:/path/with/backslashes"
+ * ```
+ */
+export function normalizePath(path: string) {
+	return path.startsWith('\\\\?\\') ? path : path.replace(/\\/g, '/');
+}
+
+/**
+ * Read and parse a JSON file.
+ *
+ * @param path File path to try and parse as JSON.
+ * @returns Parsed JSON file.
+ */
+export async function readJsonFile<T = object>(path: string) {
```

Comment:
Instead of object and this sort of generic typings what would you think of using a `JSONValue` type like this? https://dev.to/ankittanna/how-to-create-a-type-for-complex-json-object-in-typescript-d81

Reply:
It can be changed from `object` to something more inclusive like in that article if you would like, sure. My intention behind it being generic is that it overrides the type inside this function, instead of doing `as ...` each time we read a json file, so I would like to keep the generic part. If that's okay with you, that is?

In the future, the generic could ideally be changed to use the inferred type from a validator function passed in if you opt to add file validation at some point.

Reply:
Ah no yeah you're right, anyways sorry I wasn't suggesting to make the function not generic (although yeah my comment can make it look like that's what I meant 😓) but to do something like:
```
export async function readJsonFile<T extends JSONValue>(path: string) {
```
to make it less generic in the sense that we restrict what the type can be (so that for example we get compilation errors if we try to get something not JSON serializable like a date)

It's no big deal either way and as you mentioned if we had validation we could make it even quite specific, so I am fine with whatever here 🙂